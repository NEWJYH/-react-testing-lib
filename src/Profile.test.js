import React from "react";
import {render} from "@testing-library/react";
import Profile from "./Profile";

describe("<Profile />", () => {
    it("matches snapshot", () => {
        const utils = render(<Profile username="NEWJYH" name="정영훈"/>);
        expect(utils.container).toMatchSnapshot();
    });
    it("shows the props correctly", () => {
        const utils = render(<Profile username="NEWJYH" name="정영훈"/>);
        // utils.getBytText
        utils.getByText("NEWJYH"); //랜더 된 결과에서 찾아봄
        utils.getByText("(정영훈)"); // 랜더 된 결과에서 문자가 있는 지 확인하는 로직
        // utils.getByText("NEWJYH (정영훈)"); // 텍스트가 알아서 검색됨 
    });
});