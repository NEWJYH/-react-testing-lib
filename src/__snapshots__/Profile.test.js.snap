// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      NEWJYH
    </b>
     
    <span>
      (
      정영훈
      )
    </span>
  </div>
</div>
`;
